### PR TITLE
HIP-584: modify `HtsPrecompiledContract` to work with write operations from dynamic context

### PR DESCRIPTION
**Pull request by 0xivanov**
**Description**:
We need to support nested redirect operations like `ABI_ID_ERC_APPROVE` , `ABI_ID_ERC_SET_APPROVAL_FOR_ALL`,
`ABI_ID_ERC_TRANSFER`.  

In this PR I modify `HtsPrecompiledContract` to recognise write erc function selectors properly.

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java [hunk 1]
```diff
@@ -58,9 +58,10 @@
 import org.hyperledger.besu.evm.precompile.PrecompiledContract.PrecompileContractResult;
 
 /**
- * This class is a modified copy of HTSPrecompiledContract from hedera-services repo. Additionally,
- * it implements an adapter interface which is used by {@link com.hedera.mirror.web3.evm.store.contract.precompile.MirrorHTSPrecompiledContract}.
- * In this way once we start consuming libraries like smart-contract-service it would be easier to
+ * This class is a modified copy of HTSPrecompiledContract from hedera-services repo. Additionally, it implements an
+ * adapter interface which is used by
+ * {@link com.hedera.mirror.web3.evm.store.contract.precompile.MirrorHTSPrecompiledContract}. In this way once we start
+ * consuming libraries like smart-contract-service it would be easier to
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java:64]: It seems we have deleted part of the javadoc probably in resolving merge conflicts.

We can complete it with:

"delete the code base inside com.hedera.services package"

Comment (0xivanov) [reply hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java]: Added

**Review by steven-sheehy — APPROVED** [review]
LGTM. Can merge once Ivan approves

**Review by IvanKavaldzhiev — APPROVED** [review]
LGTM

**Pull request merged**